platform/pulp/platform/models/task.py

```diff
@@ -0,0 +1,231 @@
+
+
+class Worker(Model):
+    """
+    Represents a worker
+
+    Fields:
+
+    :cvar name: The name of the worker, in the format "worker_type@hostname"
+    :type name: models.TextField
+
+    :cvar last_heartbeat: A timestamp of this worker's last heartbeat
+    :type last_heartbeat: models.DateTimeField
+    """
+    name = models.TextField(db_index=True, unique=True)
+    last_heartbeat = models.DateTimeField()
+
+
+class TaskLock(Model):
+    """
+    Locking mechanism for services that utilize active/passive fail-over
+
+    Fields:
+
+    :cvar name: The name of the item that has the lock
+    :type name: models.TextField
+
+    :cvar timestamp: The time the lock was acquired
+    :type timestamp: models.DateTimeField
+
+    :cvar lock: The name of the lock acquired
+    :type lock: models.TextField
+
+    """
+    CELERY_BEAT = 'CeleryBeat'
+    RESOURCE_MANAGER = 'ResourceManager'
+    LOCK_STRINGS = (
+        (CELERY_BEAT, 'Celery Beat Lock'),
+        (RESOURCE_MANAGER, 'Resource Manager Lock')
+    )
+
+    name = models.TextField(db_index=True, unique=True)
+    timestamp = models.DateTimeField(auto_now_add=True)
+    lock = models.TextField(unique=True, null=False, choices=LOCK_STRINGS)
+
+
+class Task(Model):
+    """
+    Represents a task
+
+    Fields:
+
+    :cvar task_type: The type of task this is
+    :type task_type: models.TextField
+
+    :cvar task_group: The group this task belongs to
+    :type task_group: models.UUIDField
+
+    :cvar state: The state of the task
+    :type state: models.TextField
+
+    :cvar started_at: The time the task started executing
+    :type started_at: models.DateTimeField
+
+    :cvar finished_at: The time the task finished executing
+    :type finished_at: models.DateTimeField
+
+    :cvar error: Collection of errors that might have occurred while task was running
+    :type error: models.JSONField
+
+    :cvar result: Return value of the task
+    :type result: models.JSONField
+
+    Relations:
+
+    :cvar parent: Task that spawned this task (if any)
+    :type parent: models.ForeignKey
+
+    :cvar worker: The worker that this task is in
+    :type worker: models.ForeignKey
+    """
+
+    WAITING = 'waiting'
+    SKIPPED = 'skipped'
+    ACCEPTED = 'accepted'
+    RUNNING = 'running'
+    SUSPENDED = 'suspended'
+    FINISHED = 'finished'
+    ERROR = 'error'
+    CANCELED = 'canceled'
+    STATES = (
+        (WAITING, 'Waiting'),
+        (SKIPPED, 'Skipped'),
+        (ACCEPTED, 'Accepted'),
+        (RUNNING, 'Running'),
+        (SUSPENDED, 'Suspended'),
+        (FINISHED, 'Finished'),
+        (ERROR, 'Error'),
+        (CANCELED, 'Canceled')
+    )
+    type = models.TextField()
+    group = models.UUIDField()
+    state = models.TextField(choices=STATES)
+
+    started_at = models.DateTimeField()
+    finished_at = models.DateTimeField()
+
+    error = JSONField()
+    result = JSONField()
+
+    parent = models.ForeignKey("Task", null=True)
```

Comment:
From the look of the docs, this relationship can be [followed backwards](https://docs.djangoproject.com/en/1.8/topics/db/queries/#following-relationships-backward) i.e. the parent will have a task_set value that you can use to get the children (and somewhere else in the code can be used to convert this into something useful for the api)  We should be able to rename it via related_name to something more intuitive